.github/workflows/lint-and-check-types.yml

```diff
@@ -7,10 +7,15 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
+      - uses: pnpm/action-setup@v2
+        with:
+          version: '7'
       - uses: actions/setup-node@v3
         with:
           node-version: '18'
+          cache: 'pnpm'
       - name: Install deps and build (with cache)
-        uses: bahmutov/npm-install@v1
-      - run: yarn eslint . --max-warnings 0
-      - run: yarn tsc
+        run: pnpm install
+      - run: pnpm eslint . --max-warnings 0
+      - run: pnpm tsc
+      - run: pnpm tsc --project tsconfig.json --noEmit
```

Comment:
compiles based on the tsconfig file taken from example
https://github.com/karlhorky/archive-webpage-browser-extension/blob/main/.github/workflows/lint-check-types-and-build.yml